packages/core-compat-api/src/convertLegacyPageExtension.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  getComponentData,
+  RouteRef as LegacyRouteRef,
+} from '@backstage/core-plugin-api';
+import {
+  ExtensionDefinition,
+  PageBlueprint,
+} from '@backstage/frontend-plugin-api';
+import kebabCase from 'lodash/kebabCase';
+import { convertLegacyRouteRef } from './convertLegacyRouteRef';
+import { ComponentType } from 'react';
+import React from 'react';
+import { compatWrapper } from './compatWrapper';
+
+/** @public */
+export function convertLegacyPageExtension(
+  LegacyExtension: ComponentType<{}>,
+  overrides?: {
+    name?: string;
+    defaultPath?: string;
+  },
+): ExtensionDefinition<any> {
+  const element = <LegacyExtension />;
+
+  const extName = getComponentData<string>(element, 'core.extensionName');
+  if (!extName) {
+    throw new Error('Extension has no name');
+  }
+
+  const mountPoint = getComponentData<LegacyRouteRef>(
+    element,
+    'core.mountPoint',
```

Comment:
Do we need to worry about the semantics of `core.gatherMountPoints` too? 🤔 i don't know these things well enough

Reply:
Not for the root pages iirc, although see the discussion in the PR body for some other potential patterns. These converted page extensions will always be flat, as in there are not childrens to discover routes from.